src/mobu/user.py

```diff
@@ -21,12 +24,37 @@ class User:
     uidnumber: int
     token: str
 
-    def __init__(self, username: str, uidnumber: int):
-        self.username = username
-        self.uidnumber = uidnumber
-        self.generate_token()
+    @classmethod
+    async def create(cls, username: str, uidnumber: int) -> User:
+        token = await cls.generate_token(username, uidnumber)
+        return cls(username=username, uidnumber=uidnumber, token=token)
+
+    @classmethod
+    async def generate_token(cls, username: str, uidnumber: int) -> str:
+        if not Configuration.gafaelfawr_token:
+            return cls.generate_legacy_token(username, uidnumber)
+
+        token_url = f"{Configuration.environment_url}/auth/api/v1/tokens"
+        admin_token = Configuration.gafaelfawr_token
+        async with ClientSession() as s:
+            r = await s.post(
+                token_url,
+                headers={"Authorization": f"bearer {admin_token}"},
+                json={
+                    "username": username,
+                    "token_type": "user",
+                    "token_name": f"mobu {str(int(time.time()))}",
+                    "scopes": ["exec:notebook"],
```

Comment:
we might need to add more scopes here, since I think this token wll also be used for things like TAP queries from the notebook, but this is fine for now

Reply:
Want me to make this a configuration parameter? It wouldn't be too hard to do while I'm in there, and then we could plumb it through the Helm chart and add whatever scopes we need via `values.yaml`.

Reply:
I think it's fine for now, although we probably want to be able later on to configure it from the body of the POST that contains other options (like the URL or the name of the image) since we might want a bit of a mix of scopes going on.  Like for monkeys just running against TAP directly.  Mostly just wondering if this will maybe cause a downstream failure that I have to watch for, but it'll probably be fine.  I think this token won't be used anymore anyway with nublado2, since nublado2 I think will get another token and use that instead?

Reply:
Oh, yeah, that makes sense. I'll go with this for now.

This is the token that will be used with nublado2. mobu gets a token for mobu that has the `admin:token` scope, which allows it to mint arbitrary tokens, and this is the code where it mints a token for a user to use to run the job. So it has to say what scopes that user token is supposed to have.